src/tree/utils/mergeOneOrAnyOf.ts

```diff
@@ -0,0 +1,19 @@
+import { JSONSchema4 } from 'json-schema';
+
+export function mergeOneOrAnyOf(schema: JSONSchema4, combiner: 'oneOf' | 'anyOf'): JSONSchema4[] {
+  const merged: JSONSchema4[] = [];
+
+  if (!Array.isArray(schema.allOf)) return merged;
+
+  const items = schema[combiner];
+
+  if (Array.isArray(items)) {
+    for (const item of items) {
+      merged.push({
+        allOf: [item, ...schema.allOf],
```

Comment:
I suggest putting the `allOf` properties first, since more often than not things like `id` and other common props will be in it. Generally it's nice to show those at the top.

For example, here is redoc - they append the allOf and thus props like `id` are found in the middle of the schema, rather than at the top. Having `id` at top would be nicer IMHO.

<img width="834" alt="Screen Shot 2020-10-02 at 6 35 55 PM" src="https://user-images.githubusercontent.com/847542/94974934-bb309300-04d5-11eb-8d67-c3b19fcf2cea.png">

Reply:
Makes sense! Will do.